src/kbmod/regionsearch/region_search.py

```diff
@@ -0,0 +1,261 @@
+    - ``index``: Index of the observation.
+
+    All rows occur uniquely in the collection. The columns pointing, time, and fov are required. The columns index and distance_interval may either both be None or both not None.
+    The columns observatory_location, data_source and observation_id may be None.
+    If observatory_location is None then the observation is assumed to be geocentric.
+
+    The table is initialized in the __init__ or by calling the add method one or more times. The get_table method returns the data as an astropy.table and freezes the RowSet so that no more rows can be added.
+    Completely empty columns are left in the table with all None entries in a Column instance. This is done so that indexing columns and rows always returns data (even if just None).
+
+    """
+
+    names_type1 = ["pointing", "time", "fov", "distance_near", "distance_far", "observatory_location"]
+    names_type2 = ["data_source", "observation_id", "index"]
+    names = names_type1 + names_type2
+    dtype = [SkyCoord, Time, Quantity, Quantity, Quantity, EarthLocation, "str", "uint32", "uint32"]
+    units = [None, None, u.deg, u.au, u.au, None, None, None, None]
+
+    def __init__(self):
+        """Initializes the RowSet object."""
+        self.table = None
+        self.pointing = None
+        self.time = None
+        self.fov = None
+        self.distance_far = None
+        self.distance_near = None
+        self.observatory_location = None
+        self.data_source = None
+        self.observation_id = None
+        self.index = None
+        self.rowcnt = 0
+        self.dtype = RowSet.dtype.copy()
+
+    def add(self, **kwargs):
+        """
+        Adds a row to the RowSet.
+
+        The parameters are optional but if present must be the parameter type.
+
+        Parameters
+        ----------
+        pointing : SkyCoord
+            The pointing of the observation.
+        time : Time
+            The time of the observation.
+        fov : Quantity
+            The field of view of the observation.
+        distance_far : Quantity
+            The far distance of the observation.
+        distance_near : Quantity
+            The near distance of the observation.
+        observatory_location : EarthLocation
+            The location of the observatory.
+        data_source : str
+            The data source of the observation.
+        observation_id : int
+            The observation id of the observation.
+        """
+        if self.table is not None:
+            raise RuntimeError("Cannot add rows to a RowSet that has been converted to a table")
+
+        for k in self.names_type1:
+            if k in kwargs:
+                if getattr(self, k) is None:
+                    setattr(self, k, kwargs[k])
+                else:
+                    setattr(self, k, kwargs[k].insert(0, getattr(self, k)))
+                self.rowcnt = max(self.rowcnt, len(getattr(self, k)))
+                del kwargs[k]
```

Comment:
We can revisit should RowSet resurface.
Because you *should* remove arguments as you consume when using cooperative inheritance so they do not get reused should someone make the mistake of reusing a keyword in multiple mixins. I agree that python is super clumsy on this point. The clean way of doing this is explicit keyword arguments but that was problematic in RowSet. Again moot for now.